File: go/gin/internal/modules/address/entity/entity.go
package entity

import (
	"github.com/google/uuid"
)

type AddressEntity struct {
	ID              *uuid.UUID     `json:"id,omitempty"`
	Logradouro      string     		`json:"logradouro"`
	Numero          string     		`json:"numero"`
	Bairro          string     		`json:"bairro"`
	Cidade          string     		`json:"cidade"`
	UF              string     		`json:"uf"`
	Complemento     *string    		`json:"complemento,omitempty"`
	CEP             string     		`json:"cep"`
	IDCliente       *uuid.UUID     `json:"idCliente,omitempty"`
	DataDeCriacao   string  		`json:"dataDeCriacao"`
	DataDeAtualizacao string 	`json:"dataDeAtualizacao"`
}

File: python/flask/src/app/modules/addresses/entity.py
from datetime import datetime
from uuid import UUID, uuid4
from pydantic import BaseModel, ConfigDict, field_serializer

class AddressEntity(BaseModel):
  id: UUID = uuid4()
  logradouro: str
  numero: str
  bairro: str
  cidade: str
  uf: str
  complemento: str | None;
  cep: str
  idCliente: UUID
  dataDeCriacao: datetime
  dataDeAtualizacao: datetime

  model_config = ConfigDict(
      json_encoders={UUID: str},  # Garante que UUID seja string no JSON
      ser_json_timedelta="iso8601",  # Garante ISO 8601 para timedelta (se houver)
      ser_json_bytes="utf8",  # Se houver bytes, serializa como UTF-8
      ser_json_path="unix",  # Serializa paths corretamente
  )

  @field_serializer("dataDeCriacao", "dataDeAtualizacao")
  def serialize_datetime(self, value: datetime) -> str:
      """Garante que datetime seja sempre ISO 8601"""
      return value.isoformat()

  def to_json(self):
    return self.model_dump(exclude_none=True)